### PR TITLE
Document machine-to-machine auth options in openeo

### PR DESCRIPTION
**Pull request by soxofaan**
ref: eu-cdse/openeo-cdse-infra#233

### CODE REVIEW

File: APIs/openEO/authentication.qmd [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+-   The client credentials only identify your (OAuth) client, not your personal user account.
+
+    -   This means that openEO resources such as openEO batch jobs, their results, UDP's, etc
+        from one identity are not available to the other.
+        For example, you can not list the batch jobs created with client credentials,
+        when you are using your personal account.
+    -   Likewise, the balances of processing credits are separate.
+        It is however possible to link the balance of the client credentials account
+        to your personal account.
+        To enable that: contact support and provide your client id and user id.
```
Comment (soxofaan) [inline APIs/openEO/authentication.qmd:218]: @JanssenBrm  @rahmandawibowo-vito  this probably needs some more work. I'm not sure what you need nowadays at the marketplace side of things to link client credential accounts with user accounts

File: APIs/openEO/authentication.qmd [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+---
+
+While basic discovery of openEO collections and processes is possible without authentication, executing openEO workflows requires user authentication. This is necessary to manage user quotas, resources, and credit consumption effectively. User authentication in openEO is handled with the OpenID Connect protocol (often abbreviated as "OIDC").
+
+The openEO endpoint of the Copernicus Data Space Ecosystem is configured to use the identity provider service of the Copernicus Data Space Ecosystem. It is therefore recommended to complete the [Copernicus Data Space Ecosystem registration](../../Registration.qmd) before attempting to authenticate with openEO.
+
+
+
+
+## Getting Started: A Typical openEO Authentication Flow With The openEO Python Client Library {#sec-typical-authentication-flow}
```
Comment (Pratichhya) [inline APIs/openEO/authentication.qmd:12]: Doesn't the sub-title seem to be a bit long?

File: APIs/openEO/authentication.qmd [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+
+## Alternative Authentication Methods
+
+The [openEO Python client library documentation](https://open-eo.github.io/openeo-python-client/auth.html){target="_blank"}
+has a more in-depth information of various authentication concepts,
+and discusses alternative authentication methods.
+
+
+## Non-interactive And Machine-to-Machine Authentication with the openEO Python client library
```
Comment (Pratichhya) [inline APIs/openEO/authentication.qmd:65]: This too seems a bit longer title to me

**Review by Pratichhya — APPROVED** [review]
I assume as a first verison of the document, it is good to be merged

**Comment by Pratichhya** [comment]
@KanerLev @Slegersj @bomansb FYI

File: APIs/openEO/authentication.qmd (hunk 3 above)
Comment (soxofaan) [reply APIs/openEO/authentication.qmd:65]: To make it easy for people to find this I wanted to include "non-interactive" and "machine-to-machine" in the title. It also focuses heavily on python use cases, so I wanted to make that clear

Comment (soxofaan) [reply APIs/openEO/authentication.qmd]: made it bit shorter

File: APIs/openEO/authentication.qmd (hunk 2 above)
Comment (soxofaan) [reply APIs/openEO/authentication.qmd]: made it bit shorter

**Comment by soxofaan** [comment]
> Minor changed in the sentences with regards to pronoun "you" has been done

@Pratichhya just curious: what is the background of removing "you"? is that a style guide?


I also noticed you removed quite some newlines from my text, is that intentional?
Note that I intentionally use newlines in docs, following https://sembr.org/

**Comment by Pratichhya** [comment]
> > Minor changed in the sentences with regards to pronoun "you" has been done
> 
> @Pratichhya just curious: what is the background of removing "you"? is that a style guide?

@soxofaan, the reason to update sentences having "you" or "your" as pronouns was to maintain the idea adopted in PR #533 as suggested by @HansVRP. To limit the use of "you" instead, use "user" for consistent and academic writing.  


> I also noticed you removed quite some newlines from my text, is that intentional? Note that I intentionally use newlines in docs, following https://sembr.org/

Ahh, I am extremely sorry about the newlines style. I didn't know they were intentional and thought the markdown anyway read them as a single line. I will try to update these 😅

**Comment by soxofaan** [comment]
> the reason to update sentences having "you" or "your" as pronouns was to maintain the idea adopted in PR https://github.com/eu-cdse/documentation/pull/533 as suggested by @HansVRP. To limit the use of "you" instead, use "user" for consistent and academic writing.

Ok good to know, makes sense to add this to the guidelines (#612) I guess

**Comment by soxofaan** [comment]
> thought the markdown anyway read them as a single line

Yes that's the point: from the standpoint of Markdown tools they are considered a single paragraph. 
But the separate lines are much friendlier to other tools like git, GitHub pull requests, ...

**Comment by Pratichhya** [comment]
> > thought the markdown anyway read them as a single line
> 
> Yes that's the point: from the standpoint of Markdown tools they are considered a single paragraph. But the separate lines are much friendlier to other tools like git, GitHub pull requests, ...

Noted, thank you for the suggestion. I updated the changed sentences in the PR and will try to implement this in my other PRs too :)

**Comment by soxofaan** [comment]
Are there any open issues here I have to address or other todo's to get this merged?

**Comment by KanerLev** [comment]
Hi @soxofaan, I couldn't write earlier. Content-wise all OK from our side however to get it merged now please commit them directly to publish (not staging). 

With the current operations flow we follow:
-all PRs are committing changes directly to publish e.g. see https://github.com/eu-cdse/documentation/pulls.
-staging is rebased afterwards

We are working with all stakeholders to introduce committing to staging before carrying them to public : 
i.e. - we commit to staging during a timeframe e.g. for a week
- e.g. once in every week we commit what's accumulated in staging to public

however we encounter several bottlenecks and still need time on them:
1) on the timeframe (some changes are asked to be public within hours/days) 
2) on the technical level: staging and the staff needs to be prepared for such flow.

**Review by KanerLev — COMMENTED** [review]
Comment provided

**Comment by soxofaan** [comment]
Ok I'll rebase against the publish branch

However note that I expliclity chose staging because it is documented that way in the [README](https://github.com/eu-cdse/documentation/blob/publish/README.md):
https://github.com/eu-cdse/documentation/blob/fe8c3b0443496468a2b4259a7e49c044a698b7d9/README.md?plain=1#L16-L17

What is also not clear:
>  please commit them directly to publish (not staging).

do you really mean I should commit to the `publish` branch? Or do you mean that I should work in feature branch, branching off of `publish` and create a pull request targetting `publish`?

**Comment by KanerLev** [comment]
> Ok I'll rebase against the publish branch
> 
> However note that I expliclity chose staging because it is documented that way in the [README](https://github.com/eu-cdse/documentation/blob/publish/README.md):
> 
> https://github.com/eu-cdse/documentation/blob/fe8c3b0443496468a2b4259a7e49c044a698b7d9/README.md?plain=1#L16-L17
> 
> What is also not clear:
> 
> > please commit them directly to publish (not staging).
> 
> do you really mean I should commit to the `publish` branch? Or do you mean that I should work in feature branch, branching off of `publish` and create a pull request targetting `publish`?

On README: Indeed, thank you for highlighting, I'll edit 
- [README](https://github.com/eu-cdse/documentation/blob/publish/README.md) and 
- other resources that needs to be edited in this scope e.g. https://github.com/eu-cdse/documentation/blob/publish/EU-CDSE_Documentation_Guidelines_v0.pdf

On "commit them directly to publish (not staging)": Thanks for asking constructively, I meant the second option ("_that I should work in feature branch, branching off of publish and create a pull request targetting publish_").

**Comment by soxofaan** [comment]
I rebased this pull request targeting the `publish` branch in PR #624

**Pull request closed without merging**